source/whats-new.txt

```diff
@@ -82,6 +82,76 @@ The 3.0 driver release includes the following new features:
   `TimeOnly Struct. <https://learn.microsoft.com/en-us/dotnet/api/system.timeonly?view=net-6.0>`__
   API reference page on MSDN.
 
+- Adds support for client-side projection when using the ``Find()`` method,
+  the ``Select()`` method, or the ``Project()`` aggregation stage.
+  In previous versions of the driver, you could perform client-side
+  projection only after calling the ``ToEnumerable()`` or ``AsEnumerable()`` method, and
+  only within the ``Select()`` method.
+  
+  To learn how to enable and use client-side projection for a driver method, select the
+  corresponding tab:
+
+  .. tabs::
+
+    .. tab:: Find() Method
+       :tabid: find-method
+
+       .. code-block:: csharp
+
+          // Enable client-side projection
+          var findOptions = new FindOptions;
+          findOptions.TranslationOptions = new ExpressionTranslationOptions
+          {
+              EnableClientSideProjections = true
+          };
+         
+          var find = collection
+              .Find(doc => doc.Id == 1, findOptions);
+              .Project(doc => new { R = MyFunction(doc.Name) });
+  
+    .. tab:: Select() Method
+       :tabid: select-method
+
+       .. code-block:: csharp
+
+          // Enable client-side projection
+          var translationOptions = new ExpressionTranslationOptions
+          {
+              EnableClientSideProjections = true
+          };
+         
+          var queryable = collection
+              .AsQueryable(translationOptions)
```

Comment:
This is subtle but the actual argument to `AsQueryable` is an `AggregateOptions` value, not an `ExpressionTranslationOptions` value.

We might want to be explicit about that.

`AsQueryable(translationOptions)` does work but only because there is an implicit conversion from `ExpressionTranslationOptions` to `AggregateOptions`, and I'm not sure we're going to keep that implicit conversion.